src/Wizard/Wizard.jsx

```diff
@@ -0,0 +1,190 @@
+import classnames from "classnames";
+import React, {PropTypes} from "react";
+import _ from "lodash";
+
+import WizardStep from "./WizardStep";
+import {Button, ProgressBar} from "../";
+
+import "./Wizard.less";
+
+const INITIAL_STATE = {
+  currentStep: 0,
+  stepsVisited: [0],
+};
+export class Wizard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = _.assign(INITIAL_STATE, {
+      data: props.initialWizardData || {},
+    });
+    this.reset = this.reset.bind(this);
+    this.prevStepHandler = this.prevStepHandler.bind(this);
+    this.nextStepHandler = this.nextStepHandler.bind(this);
+  }
+
+  reset() {
+    this.setState(_.assign({}, INITIAL_STATE, {data: {}}));
+  }
+
+  prevStepHandler() {
+    const prevStep = Math.max(0, this.state.currentStep - 1);
+    this.setState({
+      currentStep: prevStep,
+      stepsVisited: _.union(this.state.stepsVisited, [prevStep]),
+    });
+  }
+
+  nextStepHandler() {
+    if (this.state.currentStep === this.props.steps.length - 1) {
+      this.props.onComplete(this.state.data);
+      return;
+    }
+    const nextStep = Math.min(this.state.currentStep + 1);
+    this.setState({
+      currentStep: nextStep,
+      stepsVisited: _.union(this.state.stepsVisited, [nextStep]),
+    });
+  }
+
+  render() {
+    const {
+      className, style, title, description, help, wizardButtons, steps, nextButtonValue,
+      prevButtonValue, seekable,
+    } = this.props;
+
+    const classes = classnames("Wizard", className);
+    const curStep = steps[this.state.currentStep];
+    const validSteps = steps.map(step => step.validate(this.state.data));
+    const percentComplete = _.compact(validSteps).length / steps.length;
+
+    const nextDisabled = this.state.currentStep === steps.length - 1 ?
+      _.compact(steps.map(step => step.validate(this.state.data))).length !== steps.length
+    : !curStep.validate(this.state.data);
```

Comment:
There's a lot going on here. Can we break down these lines into something more readable? Or at least simplify with `steps.filter(...).length`